packages/dd-trace/src/appsec/iast/analyzers/injection-analyzer.js

```diff
@@ -1,19 +1,33 @@
 'use strict'
 const Analyzer = require('./vulnerability-analyzer')
 const { isTainted, getRanges } = require('../taint-tracking/operations')
+const { SQL_ROW_VALUE } = require('../taint-tracking/source-types')
 
 class InjectionAnalyzer extends Analyzer {
   _isVulnerable (value, iastContext) {
     if (value) {
-      return isTainted(iastContext, value)
+      if (!isTainted(iastContext, value)) {
+        return false
+      }
+
+      return this._areRangesVulnerable(getRanges(iastContext, value))
     }
```

Comment:
this could be 
```javascript
if (value && isTainted(iastContext, value)) {
  return this._areRangesVulnerable(getRanges(iastContext, value))
}
return false
```